pkg/yurtmanager/controller/podreadyupdater/podreadyupdater_controller.go

```diff
@@ -0,0 +1,319 @@
+	newCondition := podutil.GetPodReadyCondition(newPod.Status)
+	if oldCondition == nil || newCondition == nil {
+		return false
+	}
+	if oldCondition.Status == corev1.ConditionTrue && newCondition.Status == corev1.ConditionFalse {
+		klog.V(2).Infof(Format("podReadyUpdaterController enqueue pod update event for %v", newPod.Name))
+		return true
+	}
+	return false
+}
+
+// nodePoolCreate filter events: nodePool disconnect with cloud
+func nodePoolCreate(evt event.CreateEvent) bool {
+	nodePool, ok := evt.Object.(*appsv1beta1.NodePool)
+	if !ok {
+		return false
+	}
+
+	if nodePool.Status.UnreadyNodeNum == int32(len(nodePool.Status.Nodes)) {
+		klog.V(2).Infof(Format("podReadyUpdaterController enqueue nodepool create event for %v", nodePool.Name))
+		return true
+	}
+	return false
+}
+
+// nodePoolUpdate filter events: nodePool disconnect with cloud
+func nodePoolUpdate(evt event.UpdateEvent) bool {
+	oldNodePool, ok := evt.ObjectOld.(*appsv1beta1.NodePool)
+	if !ok {
+		return false
+	}
+	newNodePool, ok := evt.ObjectNew.(*appsv1beta1.NodePool)
+	if !ok {
+		return false
+	}
+
+	if oldNodePool.Status.UnreadyNodeNum == newNodePool.Status.UnreadyNodeNum {
+		return false
+	}
+	if newNodePool.Status.UnreadyNodeNum == int32(len(newNodePool.Status.Nodes)) {
+		klog.V(2).Infof(Format("podReadyUpdaterController enqueue nodepool update event for %v", newNodePool.Name))
+		return true
+	}
+	return false
+}
+
+// +kubebuilder:rbac:groups=apps.openyurt.io,resources=nodepools,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods/status,verbs=update
+// +kubebuilder:rbac:groups=core,resources=nodes,verbs=get;list;watch;update;patch
+
+// Reconcile reads that state of the cluster for a Pod object and makes changes based on the state read
+// and what is in the Pod.Spec
+func (r *ReconcilePodReadyUpdater) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	// Note !!!!!!!!!!
+	// We strongly recommend use Format() to  encapsulation because Format() can print logs by module
+	// @kadisi
+	klog.Infof(Format("Reconcile PodReadyUpdater %s/%s", request.Namespace, request.Name))
+
+	// Fetch the PodReadyUpdater instance
+	instance := &corev1.Pod{}
+	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		klog.Errorf(Format("Fail to get Pod %v, %v", request.NamespacedName, err))
+		return reconcile.Result{}, err
+	}
+
+	if instance.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+
+	if len(instance.Spec.NodeName) == 0 {
+		return reconcile.Result{}, nil
+	}
+
+	node := &corev1.Node{}
+	err = r.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.NodeName}, node)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
```

Comment:
If node has autonomy or pod-binding annotation, we can mark the pod ready status to ready.

Reply:
This controller is only handler nodepool disconnect with cloud, the node operator is handler by webhook, not here.